### PR TITLE
Upgrade to AGP 8 & Kotlin 1.9.10

### PR DESCRIPTION
**Pull request by mattkranzler5**
- Upgraded to AGP 8.2.0
- Upgraded to Kotlin 1.9.10
- Upgraded to Java 11
- Consolidated common config in root `build.gradle.kts`

### CODE REVIEW

**Comment by Jawnnypoo** [comment]
Take a look at https://github.com/instacart/formula/pull/300 for some of the issues faced there too.

**Comment by mattkranzler5** [comment]
> Take a look at #300 for some of the issues faced there too.

Yep, I've referenced that a few times. Only outstanding issue is for some reason mockito cannot mock a formula class for a test. Trying to resolve but haven't been able to figure out a workaround yet.

**Comment by carrotkite** [comment]
### JaCoCo Code Coverage 78.89% :white_check_mark:
| Class | Covered | Meta | Status |
|:---|:---:|:---:|:---:|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: formula-android/src/test/java/com/instacart/formula/MockitoFormulaTest.kt [hunk 1]
```diff
@@ -17,7 +16,9 @@ class MockitoFormulaTest {
         assertThat(formula.type()).isEqualTo(MyFormula::class)
     }
 
-    class MyFormula : StatelessFormula<Unit, Unit>() {
+    class MyFormula : Formula<Unit, Unit, Unit>() {
+        override fun initialState(input: Unit) = Unit
+
```
[the remark this answers is not in the document]
Comment (Laimiux) [reply formula-android/src/test/java/com/instacart/formula/MockitoFormulaTest.kt:21]: I suspect this test was added to ensure that `mockFormula` would continue to work in our codebase.

File: .github/workflows/build-workflow.yml [hunk 2]
```diff
@@ -6,6 +6,11 @@ jobs:
     env:
       DANGER_GITHUB_API_TOKEN: ${{ secrets.DANGER_GITHUB_API_TOKEN }}
     steps:
+      - name: Configure JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'zulu'
+          java-version: 17
```
Comment (Laimiux) [inline .github/workflows/build-workflow.yml:13]: I wonder if this should be java 18?

**Review by Laimiux — APPROVED** [review]

File: .github/workflows/build-workflow.yml (hunk 2 above)
Comment (mattkranzler5) [reply .github/workflows/build-workflow.yml]: Good call, updated.

**Pull request merged**